### DISCUSSION TITLE
Compile for an old version of FDS

### DISCUSSION THREAD

**Discussion by lihy213** [firemodels/fds#13839] [Q&A]
Hello teams,
I downloaded the old version of FDS for development, whose hash is `0a68e80e0d999ce311428971db7c173997bf1ba0`. And I installed the oneAPI 2024 and VS 2022. There's some problem occurs when build this version.
![图片](https://github.com/user-attachments/assets/a1df9de3-ef0e-4598-8562-d960b342afeb)
But the environment for FDS6.9.1 is OK. Could you please figure out the reason and how should I solve this problem? Which version of ifort is suitable?
Regards.
LI

**Comment by mcgratta** [comment]
The first message is a clue `"*** Error: Intel compiler environment variable not defined."` My Chinese is a bit rusty, but my guess is that the command `ifort` is not recognized. 

Currently, Intel is converting from its old Fortran compiler (ifort) to a new one (ifx). If you try to compile old FDS source with the new Intel oneAPI compilers, you will probably have trouble. Is there a compelling reason for you to compile old FDS source? We are focusing on getting the latest source to compile with the latest compilers, be they Intel, Gnu, or others. I suggest you do the same.

**Reply by lihy213** [reply]
Not that important though. There's a self-developed version based on the old FDS source, so I want to try to compile it, but now it meets this problem. :)
If I'm stick to compiling this old one, does that mean I need to reconfigure the environment? Do you have any tutorials on this?

**Reply by mcgratta** [reply]
No, we have no tutorials on this. You first must install the compiling software successfully, so that all the environment variables are recognized by your computer. Then you may need to modify the makefile that is used to compile FDS.

**Comment by gforney** [comment]
Sometimes when i compile old versions of fds ( by checking out earlier
commits), i will copy scripts used to bulild fds and rename them so they
are not changed after i 'git checkout ' the old revision.
